### ISSUE TITLE
FUSE support

### ISSUE THREAD

**Issue by rjocoleman** [nlf/dhyve-os#25]
In c61d1252e2e221bad75ad1f2138b554f366eb7a9 `CONFIG_FUSE_FS` was unset. 

https://github.com/nlf/dhyve-os/blob/master/config/kernel#L2279

Are you open to a PR to bring FUSE support back? e.g

``` diff
diff --git a/config/kernel b/config/kernel
index 818ed0b..7fb71f9 100644
--- a/config/kernel
+++ b/config/kernel
@@ -2253,7 +2253,8 @@ CONFIG_FANOTIFY=y
 # CONFIG_QUOTA is not set
 # CONFIG_QUOTACTL is not set
 # CONFIG_AUTOFS4_FS is not set
-# CONFIG_FUSE_FS is not set
+CONFIG_FUSE_FS=y
CONFIG_OVERLAY_FS=y

#
# Caches
```

(I haven't preemptively sent a PR because I'm still using 2.2.0 and dlite 1.1.4, so haven't built a 3.0.0 beta to test `CONFIG_FUSE_FS`)

**Comment by nlf** [comment]
what's your use case? the only reason it was removed is because this is intended to be as slim as possible and adding fuse support to the kernel as well as the userspace tools does have a small impact. if your use case seems common enough i'm open to adding it back in

**Comment by rjocoleman** [comment]
I'm using AWS's ECS in production as my scheduler.
Persistent data volumes are a bit of a pain so I've elected to use [s3fs-fuse](https://github.com/s3fs-fuse/s3fs-fuse) because S3 is close and it's just a few user uploaded images being stored by a containerized legacy app. s3fs seems to be a reasonably popular solution to this persistence issue at the moment.

I was initially confused, as dlite is my go-to development env, when fuse didn't work in the container. I found the modprobe error about fuse not immediately obvious (though it should have been).
When I switched over to boot2docker they have this kernel module baked in so it worked.

I'm not too worried if it's an edge use case I am running successfully with a fuse enabled dhyve-os on dlite 1.1.4 but thought this could have been an oversight in that commit and others might get stuck too.
